### ISSUE TITLE
bug: mocks are not captured for integrated deps in fasthttp app

### ISSUE THREAD

**Issue by re-Tick** [keploy/go-sdk#178]
**Describe the bug**
The mocks of integrated dependency calls are not captured for fasthttp application.

**To Reproduce**
1. Install and run the keploy API server
2. Integrate the go-sdk in a fasthttp/sql sample application
3. Run the sample application on record mode
4. Make API call

**Expected behavior**
mocks should captured along with the keploy testcases.

**Additional context**
This can be because [here](https://github.com/keploy/go-sdk/blob/9d4331d1b26d21f1daf6322a015f8ba8aa5c9f4e/integrations/kfasthttp/fasthttp.go#L39) fasthttp stores the keployContext for a key of type `string` but in ProcessDep, the key used to get KeployContext from context is of `keploy.KctxType`  type [here](https://github.com/keploy/go-sdk/blob/9d4331d1b26d21f1daf6322a015f8ba8aa5c9f4e/pkg/keploy/mode.go#L72). I have to fix this issue in go-sdk.

**Comment by JammUtkarsh** [comment]
Can you shed some more light on this, I can't figure out what exactly needs to be done here.
Until now, I have understood that keploy is acting as a middleware, and the context needs to be transferred from keploy to fasthttp. A similar implementation is done in [kgin](https://github.com/keploy/go-sdk/blob/main/integrations/kgin/v1/gin-v1.go#L53). 

Can you guide me here?
Also,

>keployContext for a key of type string but in ProcessDep, the key used to get KeployContext from context is of keploy.KctxType type here

Does this part mean, keployContext needs to be of type `keploy.KctxType` while currently it's string?

**Comment by re-Tick** [comment]
> Can you shed some more light on this, I can't figure out what exactly needs to be done here. Until now, I have understood that keploy is acting as a middleware, and the context needs to be transferred from keploy to fasthttp. A similar implementation is done in [kgin](https://github.com/keploy/go-sdk/blob/main/integrations/kgin/v1/gin-v1.go#L53).
> 
> Can you guide me here? Also,
> 
> > keployContext for a key of type string but in ProcessDep, the key used to get KeployContext from context is of keploy.KctxType type here
> 
> Does this part mean, keployContext needs to be of type `keploy.KctxType` while currently it's string?

Yes, you get it right. the request context is used to transfer a object(called `KeployContext`) with a key(`KCTX` of type `keploy.KctxType`) to the wrapped dependency functions from the fastHttp middleware.

Currently, the `KCTX` const is converted to string to be used as string [here](https://github.com/keploy/go-sdk/blob/9d4331d1b26d21f1daf6322a015f8ba8aa5c9f4e/integrations/kfasthttp/fasthttp.go#L40) because versions before v1.41.0 of fasthttp takes key as string for SetUserValue. But v1.41.0 onwards key can be interface{} so we can skip the typcasting in the setContextValFast.
